src/basics.md

```diff
@@ -748,6 +749,63 @@ fn run() -> Result<()> {
 
 ```
 
+[ex-sha-digest]: #ex-sha-digest
+<a name="ex-sha-digest"></a>
+## Calculate the SHA-256 digest of a file
+
+[![ring-badge]][ring] [![data-encoding-badge]][data-encoding] [![cat-filesystem-badge]][cat-filesystem]
+
+Writes some data to a file, then calculates the SHA-256 digest of
+the file's contents using [`digest::Context`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+
+extern crate data_encoding;
+extern crate ring;
+
+use data_encoding::HEXUPPER;
+use ring::digest::{Context, SHA256};
+use std::fs::File;
+use std::io::{BufReader, Read, Write};
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Decode(data_encoding::DecodeError);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let path = "file.txt";
+
+    let mut output = File::create(path)?;
+    write!(output, "We will generate a digest of this text")?;
+
+    let input = File::open(path)?;
+    let mut reader = BufReader::new(input);
+    let mut context = Context::new(&SHA256);
+
+    loop {
```

Comment:
That's a great suggestion, I factored out a `sha256_digest_from_read` function which is a bit more copy-paste friendly, which I assume is part of the point of the cookbook. :)